mne/filter.py

```diff
@@ -112,22 +112,7 @@ def _overlap_add_filter(x, h, n_fft=None, zero_phase=True, picks=None,
         warnings.warn("FFT length is not a power of 2. Can be slower.")
 
     # Filter in frequency domain
-    h_fft = fft(np.r_[h, np.zeros(n_fft - n_h, dtype=h.dtype)])
-
-    if zero_phase:
-        # We will apply the filter in forward and backward direction: Scale
-        # frequency response of the filter so that the shape of the amplitude
-        # response stays the same when it is applied twice
-
-        # be careful not to divide by too small numbers
-        idx = np.where(np.abs(h_fft) > 1e-6)
-        h_fft[idx] = h_fft[idx] / np.sqrt(np.abs(h_fft[idx]))
```

Comment:
This was the problem. The original `h` is correct, and so is the zero-padded version. `h_fft` is the FFT of the zero-padded version. Changing the frequency components in this way will introduce non-zero values in the "zeros" portion of the IFFT of `h_fft`, which will in turn introduce artifacts in the signal.

This code originally existed to ensure that forward-backward filtering would yield the correct gain values. The correction (`sqrt`) of the gain values is now done *before* the filter is constructed, avoiding the issue.